From a5dc523e162033acfdef1e14d149aaab30d02a18 Mon Sep 17 00:00:00 2001
Message-ID: <a5dc523e162033acfdef1e14d149aaab30d02a18.1741704953.git.mjg@fedoraproject.org>
From: Michael J Gruber <mjg@fedoraproject.org>
Date: Tue, 11 Mar 2025 13:02:24 +0100
Subject: [PATCH] tests: adjust a few expected versions

The mupdf release (testing) branch for 1.25.x accumulated a few fixes
after 1.25.4, so adjust >= 1.26 to > 1.25.4 and similar.
---
 tests/test_annots.py      | 2 +-
 tests/test_pixmap.py      | 2 +-
 tests/test_textextract.py | 8 ++++----
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/tests/test_annots.py b/tests/test_annots.py
index 33f7e609..f0eb9d4a 100644
--- a/tests/test_annots.py
+++ b/tests/test_annots.py
@@ -477,7 +477,7 @@ def test_4047():
 
 def test_4079():
     path = os.path.normpath(f'{__file__}/../../tests/resources/test_4079.pdf')
-    if pymupdf.mupdf_version_tuple >= (1, 26):
+    if pymupdf.mupdf_version_tuple > (1, 25, 4):
         path_after = os.path.normpath(f'{__file__}/../../tests/resources/test_4079_after.pdf')
     else:
         # 2024-11-27 Expect incorrect behaviour.
diff --git a/tests/test_pixmap.py b/tests/test_pixmap.py
index a1d37373..1c387e10 100644
--- a/tests/test_pixmap.py
+++ b/tests/test_pixmap.py
@@ -416,7 +416,7 @@ def test_3448():
     print(f'{rms=}')
     if pymupdf.mupdf_version_tuple < (1, 24, 11):
         assert 30 <= rms < 45
-    elif pymupdf.mupdf_version_tuple < (1, 26):
+    elif pymupdf.mupdf_version_tuple <= (1, 25, 4):
         # Prior to fix for mupdf bug 708274.
         assert 1 < rms < 2
     else:
diff --git a/tests/test_textextract.py b/tests/test_textextract.py
index 6c77295a..58d4ad73 100644
--- a/tests/test_textextract.py
+++ b/tests/test_textextract.py
@@ -479,7 +479,7 @@ def test_4245():
     path_diff = os.path.normpath(f'{__file__}/../../tests/resources/test_4245_diff.png')
     pixmap_diff.save(path_diff)
     print(f'{rms=}')
-    if pymupdf.mupdf_version_tuple < (1, 26):
+    if pymupdf.mupdf_version_tuple <= (1, 25, 4):
         # Prior to fix for mupdf bug 708274.
         assert 0.1 < rms < 0.2
     else:
@@ -504,7 +504,7 @@ def test_4180():
     path_diff = os.path.normpath(f'{__file__}/../../tests/resources/test_4180_diff.png')
     pixmap_diff.save(path_diff)
     print(f'{rms=}')
-    if pymupdf.mupdf_version_tuple < (1, 26):
+    if pymupdf.mupdf_version_tuple <= (1, 25, 4):
         # Prior to fix for mupdf bug 708274.
         assert 0.2 < rms < 0.3
     else:
@@ -538,7 +538,7 @@ def test_4182():
     pixmap_diff.save(path_diff)
     rms = gentle_compare.pixmaps_rms(path_expected, pixmap)
     print(f'{rms=}')
-    if pymupdf.mupdf_version_tuple < (1, 26):
+    if pymupdf.mupdf_version_tuple <= (1, 25, 4):
         # Prior to fix for mupdf bug 708274.
         assert 3 < rms < 3.5
     else:
@@ -627,7 +627,7 @@ def test_4179():
         pixmap_diff.save(path_out_diff)
         print(f'Have saved to: {path_out_diff=}')
         print(f'{rms=}')
-        if pymupdf.mupdf_version_tuple < (1, 26):
+        if pymupdf.mupdf_version_tuple <= (1, 25, 4):
             # Prior to fix for mupdf bug 708274, our rects are rendered slightly incorrectly.
             assert 3.5 < rms < 4.5
         else:
-- 
2.49.0.rc1.488.g3223e8d7be

